### DISCUSSION TITLE
Actionbutton in R Shiny

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#615] [Q&A] [answered]
Saya memiliki kodingan seperti ini : 
<img width="547" alt="image" src="https://user-images.githubusercontent.com/106010284/230098631-4478ad6d-d7c1-4cad-9459-fd071db1facf.png">

Outputnya seperti ini :
<img width="670" alt="image" src="https://user-images.githubusercontent.com/106010284/230099012-b0024f23-0f12-40ca-973e-a4099b98b09f.png">

Saya ingin tombol "Lihat"nya itu berada di dalam kolom "kode", yang ketika diklik akan muncul kotak dialog yang berisi nilai dari kolom kode itu sendiri. 

Mohon bantuannya kak terima kasih 🙏

**Comment by Yusraf002** [comment]
Dear @yohanespm77 ,
Terima kasih atas pertanyaannya pak, tim kami akan segera coba bantu atau carikan referensi ya pak

**Comment by Yusraf002** [comment, marked as answer]
Dear @yohanespm77 ,

untuk menampilkan tabel yang lebih interaktif dan mudah di-customize kita bisa menggunakan function `renderDT` dari package `DT`.  Berikut kami berikan contoh kode penerapan :

```{r}
library(shiny)  
library(dplyr)  
library(DT)     # Memuat package DT

data <- data.frame(    
  jenis_barang = c("laptop","mouse","keyboard"), 
  lokasi = c("lemari","meja", "lantai"),        
  kode = c("123, 456, 789", "789,1011,456","789,888,999")  
)

ui <- fluidPage(  
  DTOutput("table")  # Menampilkan data dalam bentuk DTOutput
)

server <- function(input, output, session){  # Mengatur server
  
  output$table <- renderDT({  # Mengatur output yang akan ditampilkan dalam bentuk DT
    data %>%    # Memanggil data frame
      mutate(Lihat = sprintf('<button id="%s" type="button" class="btn btn-primary btn-sm action-button">Lihat</button>', row_number())) %>%  # Menambah kolom "Lihat" dengan menggunakan sprintf
      select(jenis_barang, lokasi, kode, Lihat) %>%   # Memilih kolom yang akan ditampilkan dalam DT
      datatable(escape = FALSE, options = list(dom = 't', pageLength = 10, paging = TRUE), callback = JS("table.column(4).nodes().to$().css({cursor: 'pointer'});")) %>%  # Mengatur tampilan DT, dengan menambahkan tombol Lihat, dan membuat fungsi callback ketika di-klik
      formatStyle(  # Mengatur tampilan pada kolom Lihat
        'Lihat',
        cursor = 'pointer',
        fontWeight = 'bold',
        color = 'black'
      )
  })
  
  observeEvent(input$table_cell_clicked, {   # Membuat observer untuk menampilkan modal ketika tombol Lihat di-klik
    cell <- input$table_cell_clicked
    if (!is.null(cell$row) && cell$col == 4) {
      showModal(modalDialog(
        title = "Kode",
        paste("Nilai Kode:", data$kode[cell$row]),
        easyClose = TRUE,
        footer = NULL
      ))
    }
  })
}

shinyApp(ui, server)
```
Apabila library DT tidak berjalan silhakan install package DT terlebih dahulu,

Sekian informasi dari kami, jika dirasa membantu dapat klik "mark as answer" pada comment ini. Terima kasih

Best Regards,
Algoritma Team.
